middleware/proxy/reverseproxy.go

```diff
@@ -147,14 +147,43 @@ func (p *ReverseProxy) ServeHTTP(rw http.ResponseWriter, req *http.Request, extr
 	}
 	defer res.Body.Close()
 
-	for _, h := range hopHeaders {
-		res.Header.Del(h)
-	}
+	if res.StatusCode == http.StatusSwitchingProtocols && outreq.Header.Get("Upgrade") == "websocket" {
```

Comment:
They are both suppose to send it. Generally the server won't buy it if it does have the upgrade headers, but I think you are right. We should probably check what the server is sending back as opposed to the client. I will make that change real quick